source/extensions/omni.isaac.lab/test/sensors/test_imu.py

```diff
@@ -0,0 +1,324 @@
+        prev_ang_acc_ball = torch.zeros((self.scene.num_envs, 3), dtype=torch.float32, device=self.scene.device)
+        prev_lin_acc_cube = torch.zeros((self.scene.num_envs, 3), dtype=torch.float32, device=self.scene.device)
+        prev_ang_acc_cube = torch.zeros((self.scene.num_envs, 3), dtype=torch.float32, device=self.scene.device)
+
+        for idx in range(1000):
+            # set velocity
+            self.scene.rigid_objects["balls"].write_root_velocity_to_sim(
+                torch.tensor([[1.0, 0.0, 0.0, 0.0, 0.0, 0.0]], dtype=torch.float32, device=self.scene.device).repeat(
+                    self.scene.num_envs, 1
+                )
+            )
+            self.scene.rigid_objects["cube"].write_root_velocity_to_sim(
+                torch.tensor([[1.0, 0.0, 0.0, 0.0, 0.0, 0.0]], dtype=torch.float32, device=self.scene.device).repeat(
+                    self.scene.num_envs, 1
+                )
+            )
+            # write data to sim
+            self.scene.write_data_to_sim()
+
+            # perform step
+            self.sim.step()
+            # read data from sim
+            self.scene.update(self.sim.get_physics_dt())
+
+            if idx < 1:
+                # check the imu accelerations
+                torch.testing.assert_close(
+                    self.scene.sensors["imu_ball"].data.lin_acc_b,
+                    prev_lin_acc_ball,
+                    rtol=1e-3,
+                    atol=1e-3,
+                )
+                torch.testing.assert_close(
+                    self.scene.sensors["imu_ball"].data.ang_acc_b,
+                    prev_ang_acc_ball,
+                    rtol=1e-3,
+                    atol=1e-3,
+                )
+                torch.testing.assert_close(
+                    self.scene.sensors["imu_cube"].data.lin_acc_b,
+                    prev_lin_acc_cube,
+                    rtol=1e-3,
+                    atol=1e-3,
+                )
+                torch.testing.assert_close(
+                    self.scene.sensors["imu_cube"].data.ang_acc_b,
+                    prev_ang_acc_cube,
+                    rtol=1e-3,
+                    atol=1e-3,
+                )
+
+                # check the imu velocities
+                torch.testing.assert_close(
+                    self.scene.sensors["imu_ball"].data.lin_vel_b,
+                    torch.tensor([[1.0, 0.0, 0.0]], dtype=torch.float32, device=self.scene.device).repeat(
+                        self.scene.num_envs, 1
+                    ),
+                    rtol=1e-4,
+                    atol=1e-4,
+                )
+                torch.testing.assert_close(
+                    self.scene.sensors["imu_cube"].data.lin_vel_b,
+                    torch.tensor([[1.0, 0.0, 0.0]], dtype=torch.float32, device=self.scene.device).repeat(
+                        self.scene.num_envs, 1
+                    ),
+                    rtol=1e-4,
+                    atol=1e-4,
+                )
```

Comment:
```suggestion
                # check the imu velocities
                torch.testing.assert_close(
                    self.scene.sensors["imu_ball"].data.lin_vel_b,
                    torch.tensor([[1.0, 1e-6, -self.scene.physics_dt*9.81]], dtype=torch.float32, device=self.scene.device).repeat(
                        self.scene.num_envs, 1
                    ),
                    rtol=1e-4,
                    atol=1e-4,
                )
                torch.testing.assert_close(
                    self.scene.sensors["imu_cube"].data.lin_vel_b,
                    torch.tensor([[1.0, 1e-6, -self.scene.physics_dt*9.81]], dtype=torch.float32, device=self.scene.device).repeat(
                        self.scene.num_envs, 1
                    ),
                    rtol=1e-4,
                    atol=1e-4,
                )
```

I found that the `write_root_velocity_to_sim` must be happening before the acceleration is being applied. i.e it seems to be setting the initial velocity before calculating the final velocity by integrating acceleration.

Reply:
I am not sure what you mean by that, could you elaborate please

Reply:
Oh I noticed that for that test the data.lin_vel_b is was returning [1.0, 0.0, 0.0495] instead of [1.0 0.0 0.0]. I think the `write_root_velocity_to_sim` is setting v_0 (initial velocity) and then a calculation of v_i = v_0 + a*dt. Then the data.lin_vel_b is returning something like v_i.